### PR TITLE
Add opentracing support

### PR DESCRIPTION
**Pull request by vetcher**

### CODE REVIEW

File: tracing/opentracing/context.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		return func(ctx context.Context, delivery *origin.Delivery) context.Context {
+			// If an error occurred, span will be root span
+			spanContext, _ := tracer.Extract(opentracing.TextMap, amqpReaderWriter{headers: delivery.Headers})
+			span := tracer.StartSpan(operationName, ext.RPCServerOption(spanContext))
+			return opentracing.ContextWithSpan(ctx, span)
+		}
+	}
+}
+
+type amqpReaderWriter struct {
```
Comment (cv21) [inline tracing/opentracing/context.go]: Describe purpose and the interface which that struct implements

File: tracing/opentracing/context.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	headers map[string]interface{}
+}
+
+func (w amqpReaderWriter) ForeachKey(handler func(key, val string) error) error {
+	for k, v := range w.headers {
+		s, ok := v.(string)
```
Comment (cv21) [inline tracing/opentracing/context.go:50]: Can we use ```fmt.Sprintf("%v", v)``` ?

File: tracing/opentracing/context_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		ev.Done()
+	}
+	fmt.Println("end of events")
+}
+
+func pubFunc(prefix string, from, to int, client amqp.Client, timeout time.Duration, storage *xStorage) {
+	publisher := client.Publisher(
+		amqp.PublisherBefore(
+			ContextToAMQP(tracer),
+			traceAMQPPub(tracer, "Publish function"),
```
Comment (cv21) [inline tracing/opentracing/context_integration_test.go:77]: Add comments for that beforefuncs

**Review by sas1024 — DISMISSED** [review]

**Review by cv21 — CHANGES_REQUESTED** [review]

File: tracing/opentracing/context.go (hunk 2 above)
Comment (cv21) [reply tracing/opentracing/context.go:50]: We could not because ```fmt.Sprintf()``` will could produce any value which is not under control.

**Review by cv21 — APPROVED** [review]

**Pull request merged**